### ISSUE TITLE
Nodes do not communicate over openvswitch SDN

### ISSUE THREAD

**Issue by jperville** [openshift/origin#10718]
In a 2-node openshift3 cluster, the master and the node don't seem to communicate over the openvswitch SDN, despite being on different hostsubnets and having a tun0 interface up.

I am trying to deploy a simple web application on a node, which should be exposed by the router, running on the master node (thanks to the region=infra node-selector).

My issue is similar to #4300, which is still open.
##### Version

``` sh
[root@master ~]# oc version
oc v1.2.1
kubernetes v1.2.0-36-g4a3f9c5
```
##### Steps To Reproduce
1. provision a 2-node openshift3 cluster using chef [cookbook](https://github.com/IshentRas/cookbook-openshift3); Using servicesSubnet: 172.30.0.0/16, clusterNetworkCIDR: 10.128.0.0/14, serviceNetworkCIDR: 172.30.0.0/16 and hostSubnetLength: 9
2. on the master, run `oc annotate --overwrite namespace default openshift.io/node-selector=region=infra`
3. make a new project (do not stay in the default project)
4. in the new project, deploy a simple web application there; make sure that the web application is scheduled to the node, not to the master.
5. add a route to expose the web application
6. try opening the route in a web browser
##### Current Result

Web browser displays error 503.
##### Expected Result

Web browser displays web page.
##### Additional Information

Master and node VMs are running Centos7 amd64, using origin-*-1.2.1 packages from http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin/

On the master:

``` sh
[root@master ~]# oc get pods -n platform
NAME                  READY     STATUS    RESTARTS   AGE
sidekiq-web-1-z1r95   1/1       Running   0          23m

[root@master ~]# oc get services --all-namespaces
NAMESPACE   NAME          CLUSTER-IP       EXTERNAL-IP   PORT(S)                   AGE
default     kubernetes    172.30.0.1       <none>        443/TCP,53/UDP,53/TCP     47m
default     router        172.30.219.117   <none>        80/TCP,443/TCP,1936/TCP   42m
platform    sidekiq-web   172.30.198.75    <none>        8080/TCP                  38m

[root@master ~]# oc get endpoints -n platform
NAME          ENDPOINTS         AGE
sidekiq-web   10.129.0.3:8080   25m

[root@master ~]# wget --no-proxy -O- http://10.129.0.3:8080/
--2016-08-30 16:52:55--  http://10.129.0.3:8080/
Connecting to 10.129.0.3:8080... failed: No route to host.
```

On the node (where the webapp pod lives):

``` sh
[root@node-1 ~]# wget --no-proxy -qO- http://10.129.0.3:8080/ | wc -l
264
```

More investigation on the master:

``` sh
[root@master ~]# oc get nodes
NAME      STATUS    AGE
master    Ready     36m
node-1    Ready     32m

[root@master ~]# oc get hostsubnet
NAME      HOST      HOST IP          SUBNET
master    master    192.168.56.220   10.128.0.0/23       # pingable from both nodes
node-1    node-1    192.168.56.221   10.129.0.0/23      # pingable from both nodes

[root@master ~]# oc get pods -n platform -l application=sidekiq-web --template='{{range .items}}HostIP: {{.status.hostIP}}   PodIP: {{.status.podIP}}{{end}}{{"\n"}}'
HostIP: 192.168.56.221   PodIP: 10.129.0.3

[root@master ~]# ip route get 192.168.56.221
192.168.56.221 dev enp0s8  src 192.168.56.112 
    cache 

[root@master ~]# sysctl net.ipv4.ip_forward
net.ipv4.ip_forward = 1  # forwarding is also enabled on the VM host

[root@master ~]#  systemctl status firewalld
● firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; disabled; vendor preset: enabled)
   Active: inactive (dead)

[root@master ~]# ip route
default via 10.0.2.2 dev enp0s3  proto static  metric 100 
10.0.2.0/24 dev enp0s3  proto kernel  scope link  src 10.0.2.15  metric 100 
10.128.0.0/14 dev tun0  proto kernel  scope link 
169.254.0.0/16 dev enp0s8  scope link  metric 1003 
172.30.0.0/16 dev tun0  scope link 
192.168.56.0/24 dev enp0s8  proto kernel  scope link  src 192.168.56.112  metric 100

[root@master ~]# ovs-vsctl list-br
br0
```

Notes:
- the `ip route` output does not display a route going to local pods, as described in the [documentation](https://docs.openshift.org/latest/admin_guide/sdn_troubleshooting.html#are-your-routes-correct).
- iptables default policy is ACCEPT
- pinging master tun0 IP (10.128.0.1) from the node (which has tun0 IP of 10.129.0.1) does not work and vice-versa
- I tried provisioning a second non-master node, which still cannot communicate over the SDN with the master and the first non-master node.

**Comment by knobunc** [comment]
Please get and run the debug.sh script referenced at https://docs.openshift.com/enterprise/3.2/admin_guide/sdn_troubleshooting.html#further-help

And get me the resulting tar file.  That will have all of the information we need to debug the system.

**Comment by jperville** [comment]
Hi @knobunc , you will find the resulting tar file here: [openshift-sdn-debug-2016-08-31.zip](https://github.com/openshift/origin/files/447259/openshift-sdn-debug-2016-08-31.zip).
Thank you for the help.

**Comment by jperville** [comment]
For information, my problem was due to all Centos7 VMs in my toy cluster having two IP addresses on interface ens0p8 (internal network), the first one being the static IP I requested and the second one a DHCP-provided IP that I never requested, which was force-set by network-manager on the first boot (but not on reboots) of the VM.

This is similar to issue https://github.com/mitchellh/vagrant/issues/5590 , which is supposedly fixed but still happened to me in reproducible manner.

I ended up applying the workaround in the above issue, which is to run `nmcli connection reload; systemctl restart network.service` on the first boot of the VM (and not on subsequent runs of "vagrant provision").
